src/BlockUniqidTransformer.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Class BlockUniqidTransformer.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use AMP_Block_Uniqid_Sanitizer;
+
+/**
+ * Transform uniqid-based IDs into cacheable IDs based on wp_unique_id.
+ *
+ * Random strings based on `uniqid()` should not be used to generate CSS class
+ * names and used in inline styles so that parsed CSS transient caching is not
+ * automatically disabled.
+ *
+ * Instead, `wp_unique_id()` should be used so that the IDs in the class names
+ * are more predictable and the CSS transient caching works as expected.
+ *
+ * @link https://github.com/ampproject/amp-wp/pull/6925
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.2.2
+ * @internal
+ */
+final class BlockUniqidTransformer implements Conditional, Service, Registerable {
```

Comment:
There's one more thing that this service should probably do: detect if parsed CSS transient caching is disabled, and if so re-enable it.

Or rather, this can be done in `MonitorCssTransientCaching` which already has some of this logic actually:

https://github.com/ampproject/amp-wp/blob/fe0000b7fdfbd4c36bc91976d1f0fc83d4bfc016/src/BackgroundTask/MonitorCssTransientCaching.php#L166-L171

It's just we wouldn't want to reset it every time the plugin is updated. So the current storage of a boolean value may not be sufficient for that.